packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -242,6 +247,13 @@ export interface EcsOptimizedAmiProps {
    */
   readonly generation?: ec2.AmazonLinuxGeneration;
 
+  /**
+   * The Windows Server version to use.
+   *
+   * @default none, uses Linux generation
+   */
+  readonly windowsVersion?: WindowsOptimizedVersion;
```

Comment:
Ideally, the `generation` and `windowsVersion` properties would be merged, but I didn't think it was worth the refactoring and breaking change.

Reply:
Maybe we can @deprecate `generation` in favor of `amazonLinuxGeneration` (still support `generation` for backwards compat and in the next MV we will remove)

Reply:
Thoughts? Maybe in a subsequent PR? At least update #3398 with this.

Reply:
Sorry, I didn't see your previous comment. I can add it to this PR

Reply:
Shouldn't we just replace it with an `image` union construct, to have the definition enforce the mutual exclusion?

Reply:
Yes!

Reply:
So what's up with this? Didn't we say this is gone?